### DISCUSSION TITLE
Capture and Preserve Motion Detection Frames

### DISCUSSION THREAD

**Discussion by jlpoolen** [Motion-Project/motion#1505] [General]
Enhancement Request

Referencing: https://github.com/Motion-Project/motion/blob/master/src/alg.c

I'd like to be able to capture and store the locations and dimensions of the red boxes surrounding areas of motion.  This would be a data intensive endeavor.  The purpose of collecting the analyses would be for an artificial intelligence kind of analysis using the frame size, locations, and movements to see a pattern of behavior.  Example: for a given scene, i.e. an alley, squirrels running on a fence create small frames moving at a certain speed compared with vehicles driving by, or people walking by, or people riding bicycles by.  

Of course, the context of what dimensions of the scene captured would be an important fact and the user would provide that.  The capture of this data could help "train" the analysis of what it is that is happening.

**Comment by tuxthepenguin84** [comment]
Hi @jlpoolen,

You can kind of do this now in a round about way with Conversion Specifiers on the picture_filename Configuration Option [https://motion-project.github.io/motion_config.html#picture_filename](url)

As motion is detected and written to filenames you can then parse the filenames at a later time (or immediately) and do your analysis using location, size, pixel changes, etc.
(see Conversion Specifiers: [https://motion-project.github.io/motion_config.html#conversion_specifiers](url))

I personally do this on my home camera system with some bash scripts and motion and have had pretty good luck on doing some basic detection of objects.

Update: You can store Conversion Specifiers in a DB: https://motion-project.github.io/motion_config.html#OptDetail_Database
"Motion can be compiled with MySQL, PostgreSQL or SQLite3 database support. When enabled Motion can be configured to add a record to a table in the database as specified by the sql_query. The query can contain the fields that are used and the values are given by using Conversion Specifiers for dynamic data like the filename, time, number of detected pixels etc."
